traffic_portal/app/src/common/directives/treeSelect/TreeSelectDirective.js

```diff
@@ -0,0 +1,236 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+var TreeSelectDirective = function($document) {
+    return {
+        restrict: "E",
+        templateUrl: "common/directives/treeSelect/tree.select.tpl.html",
+        replace: true,
+        scope: {
+            treeData: '=',
+            initialValue: '<',
+            handle: '@',
+            onUpdate: "&"
+        },
+        link: function(scope, element, attrs) {
+            /**
+             * Non-recursed ordered list of rows to display (before filtering)
+             * @type [TreeSelectDirective.RowData]
```

Comment:
Tuple is used here, should probably be an array (esp. given the below assignment would actually fail compilation if it were enforced).